frontend/device/android/device.lua

```diff
@@ -173,14 +173,13 @@ function Device:init()
         android.setScreenOffTimeout(-1)
     else
         local timeout = G_reader_settings:readSetting("android_screen_timeout")
-        if not timeout then return end
-        if timeout > 0 then
+        if timeout and timeout > 0 then
             -- set a custom timeout if we already have write settings permission.
             -- do not attempt to request permissions here.
             if android.canWriteSettings() then
                 android.setScreenOffTimeout(timeout)
             end
-        elseif timeout == -1 then
+        elseif timeout and timeout == -1 then
             android.setScreenOffTimeout(timeout)
```

Comment:
You don't need to check `if android.canWriteSettings()`  in the `timeout == -1` case?

Reply:
nope, wakelock doesn't need write permissions.

Reply:
the same happens if android.needsWakelocks()